### PR TITLE
Fix invalid path for services and entities resideing in non-default modules

### LINKED ISSUE #39273: [Bug]: [Project Design Diagrams] Incorrect file paths retrieved for services residing in non-default modules
### Description

$subject.

Refer the sample below where the service resides in the ZonalForecast.bal file of the ZonalForecast non-default module.

Screenshot 2023-01-17 at 10 03 35 AM

The absolute file path of this service is retrieved as follows.

Screenshot 2023-01-17 at 10 03 13 AM

This affects the go-to-source functionality of the project design diagrams generator.

### Steps to Reproduce

_No response_

### Affected Version(s)

Detected on 2201.3.1

### OS, DB, other environment details and versions

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by AzeemMuzammil**
Fixes #39273

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java [hunk 1]
```diff
@@ -39,7 +40,8 @@ public ModelGenerator(SemanticModel semanticModel, Module module) {
         this.module = module;
         Path moduleRootPath = module.project().sourceRoot().toAbsolutePath();
         if (module.moduleName().moduleNamePart() != null) {
-            moduleRootPath = moduleRootPath.resolve(module.moduleName().moduleNamePart());
+            moduleRootPath = moduleRootPath.resolve(ProjectDesignConstants.MODULES)
+                    .resolve(module.moduleName().moduleNamePart());
```
Comment (NipunaRanasinghe) [inline misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java:44]: have we tested this new logic against default module sources? cuz AFAIR `moduleNamePart` will not be null even for the default module. Isn't that the case?

**Review by NipunaRanasinghe — COMMENTED** [review]
@AzeemMuzammil we may have to send a separate PR to the dev tools repo by modifying our integration tests along with the new changes

File: misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java (hunk 1 above)
Comment (AzeemMuzammil) [reply misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java:44]: Yeah I tested the new logic with both default and non-default models, and it works as expected. and still the `moduleNamePart` is null for the default-module.

Comment (NipunaRanasinghe) [reply misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java:44]: great! 👍

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**